Core/include/Acts/EventData/TrackStateProxy.hpp

```diff
@@ -656,9 +689,36 @@ class TrackStateProxy {
   ///       to another. Use the `projector` or `effectiveProjector` method if
   ///       you want to access the matrix.
   /// @return The projector bitset
-  ProjectorBitset projectorBitset() const {
+  /// @deprecated Use projector() instead
+  [[deprecated("use projector() instead")]] ProjectorBitset projectorBitset()
+      const {
     assert(has<hashString("projector")>());
-    return component<ProjectorBitset, hashString("projector")>();
+    ProjectorMapping mapping = projectorMapping();
+    SubspaceHelper<eBoundSize> subspaceHelper(mapping, calibratedSize());
+    return subspaceHelper.projectorBitset();
+  }
+
+  ProjectorMapping projectorMapping() const {
+    assert(has<hashString("projector")>());
+    return component<ProjectorMapping, hashString("projector")>();
+  }
+
+  void setProjector(const ProjectorMapping& proj)
+    requires(!ReadOnly)
+  {
+    component<ProjectorMapping, hashString("projector")>() = proj;
+  }
+
+  void setProjector(std::span<const BoundIndices> proj)
```

Comment:
I am generally not so happy with this interface. It does not convert from a different int type even if I use a template. Maybe we should template on the full type and expect a range like type? @paulgessinger

Reply:
What do you want to be able to construct it from?

Reply:
I thought it would be nice to take any kind of integer sequence with runtime size. But maybe what @goetzgaycken suggested makes more sense https://github.com/acts-project/acts/pull/3453#discussion_r1695672293

So strictly taking `template<int N> setProjector(std::array<BoundIndices, N>)` or `std::uint8_t` and then storing it in the maximally sized `std::array` in the backend.